src/test/regress/expected/citus_local_tables_mx.out

```diff
@@ -724,5 +724,252 @@ $$);
+INSERT INTO foreign_table_test VALUES (1, 'text_test');
+CREATE EXTENSION postgres_fdw;
+CREATE SERVER foreign_server
+        FOREIGN DATA WRAPPER postgres_fdw
+        OPTIONS (host 'localhost', port :'master_port', dbname 'regression');
+CREATE USER MAPPING FOR CURRENT_USER
+        SERVER foreign_server
+        OPTIONS (user 'postgres');
+CREATE FOREIGN TABLE foreign_table (
+        id integer NOT NULL,
+        data text,
+        a bigserial
+)
+        SERVER foreign_server
+        OPTIONS (schema_name 'citus_local_tables_mx', table_name 'foreign_table_test');
+--verify
+SELECT partmethod, repmodel FROM pg_dist_partition WHERE logicalrelid = 'foreign_table'::regclass;
+ partmethod | repmodel
+---------------------------------------------------------------------
+ n          | s
+(1 row)
+
+CREATE TABLE parent_for_foreign_tables (
+    project_id integer
+) PARTITION BY HASH (project_id);
+CREATE SERVER IF NOT EXISTS srv1 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb1',host 'localhost', port '5432');
+CREATE SERVER IF NOT EXISTS srv2 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb2', host 'localhost', port '5432');
+CREATE SERVER IF NOT EXISTS srv3 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb3', host 'localhost', port '5432');
+CREATE FOREIGN TABLE foreign_partition_1 PARTITION OF parent_for_foreign_tables FOR VALUES WITH (modulus 3, remainder 0) SERVER srv1;
+CREATE FOREIGN TABLE foreign_partition_2 PARTITION OF parent_for_foreign_tables FOR VALUES WITH (modulus 3, remainder 1) SERVER srv2;
+SET citus.use_citus_managed_tables TO OFF;
+CREATE FOREIGN TABLE foreign_partition_3 PARTITION OF parent_for_foreign_tables FOR VALUES WITH (modulus 3, remainder 2) SERVER srv3;
+SELECT citus_add_local_table_to_metadata('foreign_partition_3');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+SET citus.use_citus_managed_tables TO ON;
+SELECT partmethod, repmodel FROM pg_dist_partition
+    WHERE logicalrelid IN ('parent_for_foreign_tables'::regclass, 'foreign_partition_1'::regclass, 'foreign_partition_2'::regclass, 'foreign_partition_3'::regclass);
+ partmethod | repmodel
+---------------------------------------------------------------------
+ n          | s
+ n          | s
+ n          | s
+ n          | s
+(4 rows)
+
+ALTER FOREIGN TABLE foreign_table SET SCHEMA public;
+ALTER FOREIGN TABLE public.foreign_table RENAME TO foreign_table_newname;
+ALTER FOREIGN TABLE public.foreign_table_newname RENAME COLUMN id TO id_test;
+ALTER FOREIGN TABLE public.foreign_table_newname ADD dummy_col bigint NOT NULL DEFAULT 1;
+ALTER FOREIGN TABLE public.foreign_table_newname ALTER dummy_col DROP DEFAULT;
+ALTER FOREIGN TABLE public.foreign_table_newname ALTER dummy_col SET DEFAULT 2;
+ALTER FOREIGN TABLE public.foreign_table_newname ALTER dummy_col TYPE int;
+ALTER TABLE foreign_table_test RENAME COLUMN id TO id_test;
```

Comment:
hmm, how does this work? We sometimes use ALTER TABLE and sometime ALTER FOREIGN table. 

Is that a PG feature? If so, why we needed to implement `ForeignTable_AlterObjectSchema`? Could that be a minor issue on `alter table .. set schema` implementation of Citus?

Reply:
I believe it comes down to the places where `OBJECT_FOREIGN_TABLE ` and `OBJECT_TABLE ` is used differently in PG code.  For example, rename table simply goes through the same code: https://github.com/postgres/postgres/blob/5a2832465fd8984d089e8c44c094e6900d987fcd/src/backend/commands/alter.c#L351-L357

 
When they go through the same code path, already working commands should work for foreign tables as well on Citus (maybe only adding `FOREIGN` to the command that Citus generates)

Reply:
(or `RELKIND_FOREIGN_TABLE` vs `RELKIND_RELATION`)

Reply:
there are still `ALTER TABLE` commands here. Can you make sure we convert all to `ALTER FOREIGN TABLE`?